larpandora/LArPandoraEventBuilding/LArPandoraShower/Algs/LArPandoraShowerAlg.cxx

```diff
@@ -358,6 +360,58 @@ double shower::LArPandoraShowerAlg::SpacePointPerpendicular(art::Ptr<recob::Spac
   return pos.Mag();
 }
 
+double shower::LArPandoraShowerAlg::SCECorrectPitch(double const& pitch, TVector3 const& pos,
+    TVector3 const& dir, unsigned int const& TPC) const {
+  const geo::Point_t geoPos{pos.X(), pos.Y(), pos.z()};
+  const geo::Vector_t geoDir{dir.X(), dir.Y(), dir.Z()};
+  return shower::LArPandoraShowerAlg::SCECorrectPitch(pitch, geoPos, geoDir, TPC);
+}
+double shower::LArPandoraShowerAlg::SCECorrectPitch(double const& pitch, geo::Point_t const& pos,
+    geo::Vector_t const& dir, unsigned int const& TPC) const {
+
+  if (!fSCE || !fSCE->EnableCalSpatialSCE()){
+    throw cet::exception("LArPandoraShowerALG") << "Trying to correct SCE pitch when service is not configured"
+      << std::endl;
+  }
+  // As the input pos is sce corrected already, find uncorrected pos
+  const geo::Point_t uncorrectedPos = pos + fSCE->GetPosOffsets(pos);
+  //Get the size of the correction at pos
+  const geo::Vector_t posOffset = fSCE->GetCalPosOffsets(uncorrectedPos, TPC);
+
+  //Get the position of next hit
+  const geo::Point_t nextPos = uncorrectedPos + pitch*dir;
```

Comment:
You're stepping one full unit of pitch from your current position, is that intended?  I'd naively expect that this would land you on the next wire along.  
I faintly remember that the other new SCE modules do the same calculation so I guess consistency has to be taken into consideration. 

Could you not calculate the corrected pitch by retrieving points that are +-1/2 a unit of pitch along the direction?

Reply:
Although I agree that technically it should be +/- half a wire pitch, +1 wire is used in both the [legacy](https://github.com/LArSoft/larreco/blob/2feebd82165bd0bac50996d46d04c72cfec7dce7/larreco/Calorimetry/Calorimetry_module.cc#L299-L302) and [Gnocci](https://github.com/LArSoft/larreco/blob/2feebd82165bd0bac50996d46d04c72cfec7dce7/larreco/Calorimetry/GnocchiCalorimetry_module.cc#L528) calorimetry modules. We can do something different to be technically correct, but I thought it was better to be consistent as the difference between the methods will be small anyway. I am open to making the change if people feel strongly about this.

Reply:
I think consistency is more important than a second order effect.  Something to consider would be a campaign to update all of the SCE calculations somewhere down the line.
Thanks for clarifying Ed.